src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -27,15 +27,50 @@ internal JsonElement(JsonDocument parent, int idx)
             _idx = idx;
         }
 
-        private JsonTokenType TokenType => _parent?.GetJsonTokenType(_idx) ?? JsonTokenType.None;
+        internal JsonElement(JsonNode parent)
+        {
+            _parent = parent;
+            _idx = -1;
+        }
 
+        /// <summary>
+        ///   Indicates if JSON element instance is immutable.
+        /// </summary>
+        public bool IsImmutable => _idx != -1;
+
+        private JsonTokenType TokenType
+        {
+            get
+            {
+                JsonDocument document = (JsonDocument)_parent;
```

Comment:
Ok, but should I support this property for `JsonNode` parent? (it shouldn't be called in that case)

Reply:
I'm solving the "I see exceptions in the debugger" problem :)

Reply:
Ok, thanks, I can change it to `as`. But you would suggest to implement this method for JsonNode as well, though it shouldn't be used (and maybe add debug assert)? Or keep it as it is just with changing casting?

Reply:
A Debug.Assert won't help either (it's worse, actually, the debugger display probably won't trap it).  You'd have to put an attribute on the property to not display in the debugger, which is fine to do. 

Actually, that's better than the `as` since it'll identify if you call it from the wrong scope.

```C#
[DebuggerBrowsable(DebuggerBrowsableState.Never)]
```